core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -0,0 +1,249 @@
+		if(ibisDocRef != null) {
+			ParsedIbisDocRef parsed = parseIbisDocRef(ibisDocRef, method);
+			IbisDoc ibisDoc = AnnotationUtils.findAnnotation(parsed.getReferredMethod(), IbisDoc.class);
+			if(ibisDoc != null) {
+				attribute.setDescribingElement(frankElement(
+						parsed.getReferredMethod().getDeclaringClass(), store));
+				ibisDocIntoFrankElement(ibisDoc, attribute);
+				if(parsed.hasOrder) {
+					attribute.setOrder(parsed.getOrder());
+				}
+				return;
+			}
+		}
+		IbisDoc ibisDoc = AnnotationUtils.findAnnotation(method, IbisDoc.class);
+		if(ibisDoc != null) {
+			ibisDocIntoFrankElement(ibisDoc, attribute);
+		}
+		else {
+			log.warn(String.format("No documentation available for FrankElement %s, attribute %s",
+					simpleName, attribute.getName()));
+		}
+	}
+
+	private class ParsedIbisDocRef {
+		private @Getter @Setter boolean hasOrder;
+		private @Getter @Setter int order;
+		private @Getter @Setter Method referredMethod;
+	}
+
+	private ParsedIbisDocRef parseIbisDocRef(IbisDocRef ibisDocRef, Method originalMethod) {
+		ParsedIbisDocRef result = new ParsedIbisDocRef();
+		result.setHasOrder(false);
+		String[] values = ibisDocRef.value();
+		String methodString = null;
+		if (values.length == 1) {
+			methodString = ibisDocRef.value()[0];
+		} else if (values.length == 2) {
+			methodString = ibisDocRef.value()[1];
+			try {
+				result.setOrder(Integer.parseInt(ibisDocRef.value()[0]));
+				result.setHasOrder(true);
+			} catch (Throwable t) {
+				log.warn(String.format("Could not parse order in @IbisDocRef annotation: "
+						+ Integer.parseInt(ibisDocRef.value()[0])));
+			}
+		}
+		else {
+			log.warn(String.format(String.format("Invalid @IbisDocRef annotation on method: %s.%s",
+					originalMethod.getDeclaringClass().getName(), originalMethod.getName())));
+			return null;
+		}
+		result.setReferredMethod(getReferredMethod(methodString, originalMethod));
+		return result;
+	}
+
+	private static Method getReferredMethod(String methodString, Method originalMethod) {
+		String lastNameComponent = methodString.substring(methodString.lastIndexOf(".") + 1).trim();
+		char firstLetter = lastNameComponent.toCharArray()[0];
+		String fullClassName = methodString;
+		String methodName = lastNameComponent;
+		if (Character.isLowerCase(firstLetter)) {
+			int index = methodString.lastIndexOf(".");
+			fullClassName = methodString.substring(0, index);
+		} else {
+			methodName = originalMethod.getName();
+		}
+		return getParentMethod(fullClassName, methodName);
+	}
+
+	private static Method getParentMethod(String className, String methodName) {
+		try {
+			Class<?> parentClass = Class.forName(className);
+			for (Method parentMethod : parentClass.getMethods()) {
+				if (parentMethod.getName().equals(methodName)) {
+					return parentMethod;
+				}
+			}
+			return null;
+		} catch (ClassNotFoundException e) {
+			log.warn(String.format("Super class [" + e + "] was not found!"));
```

Comment:
Het is niet 'e' die niet gevonden is, maar 'className'

Reply:
Gedaan.